ports/esp32/machine_adc.c

```diff
@@ -218,27 +213,63 @@ static mp_obj_t mp_machine_adc_block(machine_adc_obj_t *self) {
 }
 
 static mp_int_t mp_machine_adc_read(machine_adc_obj_t *self) {
+    mp_machine_adc_is_init_guard(self);
```

Comment:
Instead of putting these calls in 5 separate locations here, it would be simpler to just do it in two locations in:
- `apply_self_adc_channel_atten()`
- `madcblock_read_helper()`

That reduces code size, and keeps the code a little simpler.